File: dms-frontend/dms-frontend/src/types/transporter.ts
import type { SearchParams } from "./common";

export interface Transporter {
  transporterId: number;
  name: string;
  contactPerson?: string | null;
  mobile?: string | null;
  address?: string | null;
  agreementValidTill?: string | null;
  isActive: boolean;
  currentDriverCount: number;
}

export interface CreateTransporterRequest {
  name: string;
  contactPerson?: string;
  mobile?: string;
  address?: string;
  agreementValidTill?: string;
}

export interface UpdateTransporterRequest {
  name: string;
  contactPerson?: string;
  mobile?: string;
  address?: string;
  agreementValidTill?: string;
  isActive: boolean;
}

export interface DriverTransporterAssignment {
  assignmentId: number;
  driverId: number;
  driverName: string;
  transporterId: number;
  transporterName: string;
  assignmentDate: string;
  endDate?: string | null;
  isCurrent: boolean;
}

export interface AssignDriverRequest {
  driverId: number;
  assignmentDate?: string;
}

export interface TransporterSearchParams extends SearchParams {
  searchTerm?: string;
  isActive?: boolean;
}

File: src/Driver.Domain/Entities/Transporter.cs
using DriverDms.Domain.Common;

namespace DriverDms.Domain.Entities;

public class Transporter : BaseEntity
{
    public int TransporterId { get; set; }
    public string Name { get; set; } = default!;
    public string? ContactPerson { get; set; }
    public string? Mobile { get; set; }
    public string? Address { get; set; }
    public DateTime? AgreementValidTill { get; set; }
    public bool IsActive { get; set; } = true;

    public ICollection<DriverTransporterHistory> DriverAssignments { get; set; } = new List<DriverTransporterHistory>();
}

public class DriverTransporterHistory : BaseEntity
{
    public int AssignmentId { get; set; }
    public int DriverId { get; set; }
    public Driver? Driver { get; set; }
    public int TransporterId { get; set; }
    public Transporter? Transporter { get; set; }
    public DateTime AssignmentDate { get; set; }
    public DateTime? EndDate { get; set; }
    public bool IsCurrent { get; set; }
}
